### ISSUE TITLE
DEFAULT_DNS_SLEEP doesn't seem to do anything

### ISSUE THREAD

**Issue by szjur** [acmesh-official/acme.sh#2773]
DEFAULT_DNS_SLEEP doesn't seem to do anything, at least the only places it appears in acme.sh are:

DEFAULT_DNS_SLEEP=120

and

--dnssleep  [$DEFAULT_DNS_SLEEP]                  The time in seconds to wait for all the txt records to take effect in dns api mode. Default $DEFAULT_DNS_SLEEP seconds.

The default sleep is 20 seconds effectively (hardcoded elsewhere). Setting DEFAULT_DNS_SLEEP in account.conf not surprisingly has no effect either. The only thing that seems to work is the --dnssleep command line switch. Or am I going mad?
I recently started having renewal errors due to that default 20-second sleep - I use CNAME to a common domain and apparently Let's encrypt infra caches previous values which acme.sh dns plugin creates with 60-sec ttl.

**Comment by Neilpang** [comment]
Yes, I just removed it.
You don't need to use it in common cases.  acme.sh can poll the dns status by Doh, from cloudflare or google.

**Comment by szjur** [comment]
Well I use it with my own dns and nsupdate plugin and I have started getting authentication errors recently which I presume could be down to dns caching. That could potentially coincide with Let's Encrypt announcing the use of more servers on their side for verification. How could acme.sh possibly know what Let's Encrypt servers have in their caches? All it checks is that it looks fine from the perspective of the server it runs on and apparently there can be a drift between this and Let's Encrypt infra perspective.

As I said with CNAME type of auth pointing _acme-challenge TXT to a shared auth domain caching can be of an issue when cron renews several domains one by one. The TXT record is added with 60-sec TTL and often single domain renewal takes less. Since DEFAULT_DNS_SLEEP does nothing I started passing --dnssleep 40 to the cron job and that made it more reliable. This makes it sleep AT LEAST 40 seconds before veryfying the DNS change and in my case it seems ok combined with that 60-second TTL which is hard-coded in dns_nsupdate.sh:

update add ${fulldomain}. 60 in txt "${txtvalue}"

I hope you won't remove that --dnssleep param completely - it is useful in my setup which your docs refer to as "DNS alias mode" (https://github.com/acmesh-official/acme.sh/wiki/DNS-alias-mode).

By the way not sure where that "300" in your PR came from - is that just an example or should represent the default? The default logic in case of --dnssleep not passed is sleep 20 seconds:

[Sun Mar  1 04:25:44 CET 2020] Let's check each dns records now. Sleep 20 seconds first.
[Sun Mar  1 04:26:04 CET 2020] Checking my.domain.com for _acme-challenge.my.shared.auth.domain.com
[Sun Mar  1 04:26:05 CET 2020] Not valid yet, let's wait 10 seconds and check next one.
[Sun Mar  1 04:26:15 CET 2020] Let's wait 10 seconds and check again.
[Sun Mar  1 04:26:25 CET 2020] Checking my.domain.com for _acme-challenge.my.shared.auth.domain.com
[Sun Mar  1 04:26:26 CET 2020] Domain my.domain.com '_acme-challenge.my.shared.auth.domain.com' success.
